[FLINK-16292][build system] Add missing end to end tests

rmetzger opened the pull request:
## What is the purpose of the change

- On azure, we are using the `run-nightly-tests.sh` script to run the end to end tests. However, not all end to end tests are listed in that file. In this change, we add a first batch of missing end to end tests from the `tools/travis/splits` folder.
- Currently, all PRs / master pushes are marked red, because of failing end to end tests. This change has green e2e tests, making the overall test green.


## Brief change log

 - As a follow up to https://github.com/apache/flink/pull/11222, this PR is unifying the mechanism for switch between jdk8 and jdk11.
- The missing e2e tests from the splits folder were added

## Verifying this change

The change has been tested here: https://dev.azure.com/rmetzger/Flink/_build/results?buildId=5684&view=results

## Does this pull request potentially affect one of the following parts:

only build system files were touched

zentol commented:
Please add the appropriate component tags to the commit messages and PR title.

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -185,5 +198,32 @@ run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR
 printf "\n[PASS] All bash e2e-tests passed\n"
+
+printf "\n\n==============================================================================\n"
+printf "Running Java end-to-end tests\n"
```
add the corresponding message for bash tests

rmetzger commented:
> Please add the appropriate component tags to the commit messages and PR title.

Done

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -185,5 +198,32 @@ run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR
+mvn ${MVN_COMMON_OPTIONS} ${MVN_LOGGING_OPTIONS} ${PROFILE} verify -pl ${e2e_modules} -DdistDir=$(readlink -e build-target)
+
+
 printf "\n Please remember to run java e2e-tests by running mvn verify -Dcategories=\"\" -DdistDir=<path to flink-dist> in flink-end-to-end-tests\n"
```
can be removed

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -185,5 +198,32 @@ run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR
+printf "==============================================================================\n"
+
+set -x
+HERE="`dirname \"$0\"`"				# relative
```
reduce whitespace

zentol replied:
ping

rmetzger replied:
are you talking about the vertical whitespace?

zentol replied:
yes

rmetzger replied:
Consider it gone :)

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -173,6 +183,9 @@ run_test "ConnectedComponents iterations with high parallelism end-to-end test"
 
+run_test "Shaded Hadoop S3A with credentials provider end-to-end test" "$END_TO_END_DIR/test-scripts/test_batch_wordcount.sh hadoop_with_provider"
+
+
```
remove one empty line

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -161,69 +161,7 @@ run_test "Walkthrough Table Scala nightly end-to-end test" "$END_TO_END_DIR/test
 run_test "Walkthrough DataStream Java nightly end-to-end test" "$END_TO_END_DIR/test-scripts/test_datastream_walkthroughs.sh java"
 run_test "Walkthrough DataStream Scala nightly end-to-end test" "$END_TO_END_DIR/test-scripts/test_datastream_walkthroughs.sh scala"
 
-run_test "Test PubSub connector with Docker based Google PubSub Emulator" "$END_TO_END_DIR/test-scripts/test_streaming_gcp_pubsub.sh"
```
I know some people use this file to run things locally; so this may break someone's workflow.

rmetzger replied:
I was considering setting up another file like this for the ci runs (but then it would be even harder to keep things in sync)

Given how aggressive the end to end test scripts modify things on the file system, I doubt that many people use these scripts.
That's why I decided for this approach.

How about keeping this file, and I comment out, instead of deleting?

rmetzger replied:
I now had 2 runs where all the end to end tests where passing.
If I have another successful run, I will include all end to end tests in this PR. Stay tuned :)

zentol commented:
CI != build system. I suggest to add a new Azure component to JIRA.

rmetzger commented:
How about using the "Test Infrastructure" component for that?
I'm also not the biggest fan of having "TravisX" as a build category in our code. What if we are switching our Ci system again?
Ideally, in the code, we can call things "CIX"?

zentol commented:
Test infrastructure is for junit/maven stuff. There's little reason why, for example, some caching change for azure should be marked as such.

I'm perfectly content with having travis/azure as categories. It frees the azure category of all the baggage from the past.

rmetzger commented:
I have added a new JIRA component for Azure.

I also did a major update of this PR. The end to end tests are now green (https://dev.azure.com/rmetzger/Flink/_build/results?buildId=5707&view=results), and I think I added all end to end tests.

I would appreciate if we could merge this quickly, to finally have overall green builds again.

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -182,5 +205,29 @@ run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR
+printf "==============================================================================\n"
+
+HERE="`dirname \"$0\"`"				# relative
+HERE="`( cd \"${HERE}\" && pwd -P)`" 	# absolutized and normalized
```
This resolves to `flink/flink-end-to-end-tests`, no? (based on the code at the top, which is identical).
If so, then you are excluding all walkthrough modules since we will not be found.

rmetzger replied:
It found the walkthrough modules?!

```
2020-02-28T03:53:27.2778157Z [INFO] flink-cli-test ..................................... SUCCESS [  6.257 s]
2020-02-28T03:53:27.2779191Z [INFO] flink-parent-child-classloading-test-program ....... SUCCESS [  0.374 s]
2020-02-28T03:53:27.2780199Z [INFO] flink-parent-child-classloading-test-lib-package ... SUCCESS [  0.235 s]
2020-02-28T03:53:27.2797292Z [INFO] flink-dataset-allround-test ........................ SUCCESS [  0.234 s]
2020-02-28T03:53:27.2798579Z [INFO] flink-dataset-fine-grained-recovery-test ........... SUCCESS [  1.567 s]
2020-02-28T03:53:27.2799589Z [INFO] flink-datastream-allround-test ..................... SUCCESS [  3.847 s]
2020-02-28T03:53:27.2800549Z [INFO] flink-batch-sql-test ............................... SUCCESS [  0.272 s]
2020-02-28T03:53:27.2813243Z [INFO] flink-stream-sql-test .............................. SUCCESS [  0.304 s]
2020-02-28T03:53:27.2814296Z [INFO] flink-bucketing-sink-test .......................... SUCCESS [  3.753 s]
2020-02-28T03:53:27.2815273Z [INFO] flink-distributed-cache-via-blob ................... SUCCESS [  0.270 s]
2020-02-28T03:53:27.2816627Z [INFO] flink-high-parallelism-iterations-test ............. SUCCESS [  8.881 s]
2020-02-28T03:53:27.2830226Z [INFO] flink-stream-stateful-job-upgrade-test ............. SUCCESS [  0.903 s]
2020-02-28T03:53:27.2831268Z [INFO] flink-queryable-state-test ......................... SUCCESS [  1.892 s]
2020-02-28T03:53:27.2832421Z [INFO] flink-local-recovery-and-allocation-test ........... SUCCESS [  0.139 s]
2020-02-28T03:53:27.2835929Z [INFO] flink-elasticsearch5-test .......................... SUCCESS [  3.194 s]
2020-02-28T03:53:27.2845579Z [INFO] flink-elasticsearch6-test .......................... SUCCESS [  3.591 s]
2020-02-28T03:53:27.2846591Z [INFO] flink-quickstart-test .............................. SUCCESS [  0.269 s]
2020-02-28T03:53:27.2847947Z [INFO] flink-confluent-schema-registry .................... SUCCESS [  1.587 s]
2020-02-28T03:53:27.2858556Z [INFO] flink-stream-state-ttl-test ........................ SUCCESS [  3.828 s]
2020-02-28T03:53:27.2859600Z [INFO] flink-sql-client-test .............................. SUCCESS [  1.137 s]
2020-02-28T03:53:27.2860556Z [INFO] flink-streaming-file-sink-test ..................... SUCCESS [  0.150 s]
2020-02-28T03:53:27.2861513Z [INFO] flink-state-evolution-test ......................... SUCCESS [  0.981 s]
2020-02-28T03:53:27.2872757Z [INFO] flink-rocksdb-state-memory-control-test ............ SUCCESS [  0.950 s]
2020-02-28T03:53:27.2873920Z [INFO] flink-end-to-end-tests-common ...................... SUCCESS [  1.967 s]
2020-02-28T03:53:27.2874908Z [INFO] flink-metrics-availability-test .................... SUCCESS [ 16.337 s]
2020-02-28T03:53:27.2876415Z [INFO] flink-metrics-reporter-prometheus-test ............. SUCCESS [ 17.309 s]
2020-02-28T03:53:27.2887732Z [INFO] flink-heavy-deployment-stress-test ................. SUCCESS [  7.767 s]
2020-02-28T03:53:27.2888749Z [INFO] flink-connector-gcp-pubsub-emulator-tests .......... SUCCESS [  1.046 s]
2020-02-28T03:53:27.2889897Z [INFO] flink-streaming-kafka-test-base .................... SUCCESS [  0.403 s]
2020-02-28T03:53:27.2891588Z [INFO] flink-streaming-kafka-test ......................... SUCCESS [  7.204 s]
2020-02-28T03:53:27.2898790Z [INFO] flink-streaming-kafka011-test ...................... SUCCESS [  6.475 s]
2020-02-28T03:53:27.2900043Z [INFO] flink-streaming-kafka010-test ...................... SUCCESS [  6.407 s]
2020-02-28T03:53:27.2912375Z [INFO] flink-plugins-test ................................. SUCCESS [  0.073 s]
2020-02-28T03:53:27.2913529Z [INFO] dummy-fs ........................................... SUCCESS [  0.160 s]
2020-02-28T03:53:27.2914555Z [INFO] another-dummy-fs ................................... SUCCESS [  0.181 s]
2020-02-28T03:53:27.2921875Z [INFO] flink-tpch-test .................................... SUCCESS [  0.940 s]
2020-02-28T03:53:27.2929829Z [INFO] flink-streaming-kinesis-test ....................... SUCCESS [ 13.146 s]
2020-02-28T03:53:27.2930929Z [INFO] flink-elasticsearch7-test .......................... SUCCESS [  3.704 s]
2020-02-28T03:53:27.2932805Z [INFO] flink-end-to-end-tests-common-kafka ................ SUCCESS [02:17 min]
2020-02-28T03:53:27.2944821Z [INFO] flink-tpcds-test ................................... SUCCESS [  0.310 s]
2020-02-28T03:53:27.2945931Z [INFO] flink-walkthrough-common ........................... SUCCESS [  0.945 s]
2020-02-28T03:53:27.2947182Z [INFO] flink-walkthrough-table-java ....................... SUCCESS [  3.092 s]
2020-02-28T03:53:27.2948201Z [INFO] flink-walkthrough-table-scala ...................... SUCCESS [  0.183 s]
2020-02-28T03:53:27.2959112Z [INFO] flink-walkthrough-datastream-java .................. SUCCESS [  0.191 s]
2020-02-28T03:53:27.2960510Z [INFO] flink-walkthrough-datastream-scala ................. SUCCESS [  0.168 s]
```

rmetzger replied:
`HERE` is not used to resolve the modules. It is just used for the log4j file and the artifacts file.

rmetzger replied:
I'm not sure though if the java end to end tests are properly executed, because 4 minutes seems very short.
But I wanted to investigate that another ticket

zentol replied:
4 minutes should actually be fine.

rmetzger replied:
Ok

zentol commented on flink-end-to-end-tests/run-nightly-tests.sh:
```diff
@@ -182,5 +205,29 @@ run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR
+e2e_modules="${e2e_modules},$(find flink-walkthroughs -mindepth 2 -maxdepth 2 -name 'pom.xml' -printf '%h\n' | sort -u | tr '\n' ',')"
+
+PROFILE="$PROFILE -Pe2e-travis1 -Pe2e-travis2 -Pe2e-travis3 -Pe2e-travis4 -Pe2e-travis5 -Pe2e-travis6"
+mvn ${MVN_COMMON_OPTIONS} ${MVN_LOGGING_OPTIONS} ${PROFILE} verify -pl ${e2e_modules} -DdistDir=$(readlink -e build-target)
```
I would be surprised if this actually works given that build-target should not exist in the current directory.

zentol replied:
If the build is green regardless there's likely a but in the error reporting.

Where do you actually fail with an error if this command fails?

rmetzger replied:
This has worked in a build last night: https://dev.azure.com/rmetzger/Flink/_build/results?buildId=5709&view=logs&j=b1623ac9-0979-5b0d-2e5e-1377d695c991&t=e7804547-1789-5225-2bcf-269eeaa37447

rmetzger replied:
> Where do you actually fail with an error if this command fails?

I will fix this!

zentol commented on tools/azure-pipelines/free_disk_space.sh:
```diff
@@ -0,0 +1,35 @@
+# limitations under the License.
+
+echo "=============================================================================="
+echo "Freeing up disk space on CI system"
```
document why we do this

rmetzger commented:
I pushed an update. Is there anything outstanding?
